src/main/java/bio/terra/app/controller/DatasetsApiController.java

```diff
@@ -194,6 +197,28 @@ public ResponseEntity<SnapshotBuilderGetConceptsResponse> getConcepts(
     return ResponseEntity.ok(snapshotBuilderService.getConceptChildren(id, conceptId));
   }
 
+  @Override
+  public ResponseEntity<DatasetDataModel> queryDatasetDataById(
+      UUID id, String table, QueryDataRequestModel queryDataRequest) {
+    AuthenticatedUserRequest userReq = getAuthenticatedInfo();
+    verifyDatasetAuthorization(userReq, id.toString(), IamAction.READ_DATA);
+    // TODO: Remove after https://broadworkbench.atlassian.net/browse/DR-2588 is fixed
+    SqlSortDirection sortDirection =
+        Objects.requireNonNullElse(queryDataRequest.getDirection(), SqlSortDirection.ASC);
+    DatasetDataModel previewModel =
+        datasetService.retrieveData(
+            userReq,
+            id,
+            table,
+            queryDataRequest.getLimit(),
+            queryDataRequest.getOffset(),
+            queryDataRequest.getSort(),
+            sortDirection,
+            queryDataRequest.getFilter());
+    return ResponseEntity.ok(previewModel);
+  }
+
+  @Hidden
```

Comment:
Would it be worth it to change our `@Hidden` endpoints to redirect to their replacements?  I'm not wording this super clearly, but something like this:

```
@Hidden
@Override
  public ResponseEntity<DatasetDataModel> lookupDatasetDataById(
      UUID id,
      String table,
      Integer offset,
      Integer limit,
      String sort,
      SqlSortDirection direction,
      String filter) {
      var queryDataRequest = <construct this from the raw inputs>;
      return queryDatasetDataById(id, table, queryDataRequest);
}

Reply:
Definitely a reasonable path forward - I've gone back and forth here. 
My main hesitancy here is that I have switched all of our tests to use the post endpoint. I think it's reasonable to move our testing to the new endpoints since we're marking these old endpoints as deprecated. But, this also means that I feel less confident in making changes to these methods since we don't have the automated testing on it. 
What do you think?

Reply:
Closing the loop after discussing in stand-up -- we agreed that maintaining unit test coverage of both old and new endpoints via parameterized tests would be good enough assurance, and we can leave integration / connected tests to use the new endpoints!

Reply:
@okotsopoulos  - Thanks! I think using the method source parameterized tests were a great idea. I've made this change and I've changed the lookup endpoints to reference the new query code in the controller.